### ISSUE TITLE
ForbiddenMethodCall with Kotlin getters as property

### ISSUE THREAD

**Issue by pinguinjkeke** [detekt/detekt#5015]
## Expected Behavior
Assume we have:
Java class:
```java
package com.hello;

class Hello {
    public String getHello() {
        return "hello";
    }
}
```
Detekt configuration:
```yaml
  ForbiddenMethodCall:
    active: true
    methods:
      - 'com.hello.Hello.getHello'
```
Somewhere in Kotlin:
```kotlin
val hello = Hello()
hello.hello // no error, but must be one
hello.getHello() // error, it's correct
```

## Observed Behavior
Kotlin property-getter not caught by Detekt

## Your Environment

* Version of detekt used: 1.20.0
* Operating System and version: Mac OS, Linux

**Comment by pinguinjkeke** [comment]
@schalkms yes, sure, it won't work with Java's method w/o type resolution.

**Comment by schalkms** [comment]
What exactly is the issue with the `ForbiddenMethodCall` rule then? A property call in Kotlin, even if a Java getter/setter method is being called, can't be detected by this rule. From Kotlin's point of view, it's still a property not a method call.
In any case, I'm not sure, why accessing a getter is considered *forbidden*.

**Comment by pinguinjkeke** [comment]
@schalkms we have a 3rd-party library with unstable method which usage we're trying to disallow.

Rule description says:
This rule allows to set a list of forbidden methods. This can be used to discourage the use of unstable, experimental or deprecated methods, **especially for methods imported from external libraries**. Detekt will then report all method invocations that are forbidden.

**Comment by schalkms** [comment]
`ForbiddenProperty` sounds like a good idea.
For me, it's counterintuitive to have the `ForbiddenMethodCall` rule to flag a property.

**Comment by BraisGabin** [comment]
I'm not sure about that... You are forbidding the usage of a Java method. If we create a new rule for this you should forbid that method in two different rules.

**Comment by pinguinjkeke** [comment]
@BraisGabin that's right. We can change a behavior of `ForbiddenMethodCall` for `KProperty` that has `javaGetter` inside.

**Comment by cortinico** [comment]
> we have a 3rd-party library with unstable method which usage we're trying to disallow. It looks like a getter, but contains hard and unstable logic underneath.

@pinguinjkeke Have you considered just forbidding the imports of such library?

> ForbiddenProperty sounds like a good idea.

My 2 cents here: this sounds like a use case for a third-party rule. It's still generic, so could be reused by others, but it's not really common to forbid a property so I'm unsure if we want to include this as well in the 1st party rules.

**Comment by pinguinjkeke** [comment]
> @pinguinjkeke Have you considered just forbidding the imports of such library?

@cortinico no. The library contains a large god-class where we need to forbid some methods only. Occasionally, these methods names are starting with `get*`.

Let me tell you more about this method. Imagine a typical ORM that has a getter like: `getRelations()` which will N+1 load everything connected to an entity:
- These calls may affect performance
- We are not in the world where `get` prefix means only POJO
- Kotlin's property syntax makes it even worse: `.getRelations()` call says that it's a method and some logic may be inside, while `.relations` look small and harmless.
- Current rule implementation won't allow to completely forbid `.getRelations()` calls

> My 2 cents here: this sounds like a use case for a third-party rule. It's still generic, so could be reused by others, but it's not really common to forbid a property so I'm unsure if we want to include this as well in the 1st party rules.

Why not? Take an example from tests with `Calendar`. 

We're forbidding the `getFirstDayOfWeek()` expecting it to be an error while `firstDayOfWeek` should be fine. It's completely wrong.

Overall, all we did here is disallowed the usage of Java getter in prior of Kotlin's property call syntax. If that's what we want rule like `PreferKotlinPropertiesOverJavaGetters` should be created.

Moreover, IntelliJ IDEA's won't even highlight `getFirstDayOfWeek()` as a method out of the box, it will show it as a property and even if you will write `getFirstDayOfWeek()` manually, IDE will highlight it and suggest to change the method call to a property syntax.

From my side, the rule declaration says the following: "Forbid Calendar.getFirstDayOfWeek() method call". The ability to call `Calendar.firstDayOfWeek` violates the statement above.

BTW, some setters may be desired to be disallowed too.

**Comment by pinguinjkeke** [comment]
I think we can help with PR implementaion for this change but we need to know the way how to implement it:
- Modify `ForbiddenMethodCall`
- Create `ForbiddenProperty`

**Comment by BraisGabin** [comment]
I vote for modify `ForbiddenMethodCall`.

**Comment by cortinico** [comment]
> From my side, the rule declaration says the following: "Forbid Calendar.getFirstDayOfWeek() method call". The ability to call `Calendar.firstDayOfWeek` violates the statement above.

Thanks for the clarification. It makes sense.

Updating `ForbiddenMethodCall` sounds like the best approach then.

> BTW, some setters may be desired to be disallowed too.

Yup this goes hand in hand with the getter logic.

**Comment by pinguinjkeke** [comment]
I also discovered that the rule is not working with method references. Created a PR
